src/components/common/PaginatedTxns/index.tsx

```diff
@@ -9,93 +10,78 @@ import InfiniteScroll from '../InfiniteScroll'
+import { type TxFilter, useTxFilter } from '@/utils/tx-history-filter'
 import { BatchExecuteHoverProvider } from '@/components/transactions/BatchExecuteButton/BatchExecuteHoverProvider'
+import { isTransactionListItem } from '@/utils/transaction-guards'
 
 const NoQueuedTxns = () => (
   <Box mt="5vh">
     <PagePlaceholder imageUrl="/images/no-transactions.svg" text="Queued transactions will appear here" />
   </Box>
 )
 
-const getResultCount = (filter: TxFilter, page: TransactionListPage) => {
+const getFilterResultCount = (filter: TxFilter, page: TransactionListPage) => {
   const count = page.results.filter(isTransactionListItem).length
 
   return `${page.next ? '> ' : ''}${count} ${filter.type} transactions found`.toLowerCase()
 }
 
-const TxPage = ({
-  pageUrl,
-  useTxns,
-  onNextPage,
-  isFirstPage,
-}: {
-  pageUrl: string
-  useTxns: typeof useTxHistory | typeof useTxQueue
-  onNextPage?: (pageUrl?: string) => void
-  isFirstPage: boolean
-}): ReactElement => {
-  const { page, error } = useTxns(pageUrl)
+const PaginatedTxns = ({ useTxns }: { useTxns: typeof useTxHistory | typeof useTxQueue }): ReactElement => {
+  const [pages, setPages] = useState<Array<{ pageUrl?: string; data: TransactionListPage }>>([])
+  const [pageUrl, setPageUrl] = useState<string>()
   const [filter] = useTxFilter()
-
   const isQueue = useTxns === useTxQueue
+  const allItems = useMemo(() => pages.flatMap((page) => page.data.results), [pages])
 
-  if (page?.results) {
-    return (
-      <>
-        {/* FIXME: batching will only work for the first page results */}
-        {isFirstPage && (
-          <Box display="flex" flexDirection="column" alignItems="flex-end" mt={['-94px', '-44px']} mb={['60px', 0]}>
-            {isQueue ? <BatchExecuteButton items={page.results} /> : <TxFilterButton />}
-            {filter && <Typography mt={2}>{getResultCount(filter, page)}</Typography>}
-          </Box>
-        )}
-
-        {page.results.length ? <TxList items={page.results} /> : isQueue && <NoQueuedTxns />}
-
-        {onNextPage && page.next && (
-          <Box my={4} textAlign="center">
-            <InfiniteScroll onLoadMore={() => onNextPage(page.next)} />
-          </Box>
-        )}
-      </>
-    )
-  }
-
-  if (error) {
-    return <ErrorMessage>Error loading transactions</ErrorMessage>
-  }
-
-  return <SkeletonTxList />
-}
+  // Reset the pages when the filter changes
+  useEffect(() => {
+    setPages([])
+    setPageUrl(undefined)
+  }, [filter, useTxns])
 
-const PaginatedTxns = ({ useTxns }: { useTxns: typeof useTxHistory | typeof useTxQueue }): ReactElement => {
-  const [pages, setPages] = useState<string[]>([''])
-  const [filter] = useTxFilter()
+  // Load the current page
+  const { page: lastPage, error, loading } = useTxns(pageUrl)
 
+  // When a tx page is loaded, update the list of pages
   useEffect(() => {
-    setPages([''])
-  }, [filter])
+    if (!lastPage) return
```

Comment:
Is this meant to be the current page? The name could be confused with being the last page in the pages array i.e. pages.length - 1

Reply:
Will rename 👍